### PR TITLE
Add 'duplicate' to the set theory filters

### PR DESCRIPTION
**Pull request by flatline-studios**
##### SUMMARY
'duplicate' will take a list of values, and return all duplicate values found within it, if any.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
duplicate

##### ANSIBLE VERSION
```paste below
ansible 2.6.3
  config file = /var/ansible/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vagrant/.local/lib/python3.6/site-packages/ansible
  executable location = /home/vagrant/.local/bin/ansible
  python version = 3.6.5 (default, Apr  1 2018, 05:46:30) [GCC 7.3.0]
```

##### ADDITIONAL INFORMATION

```paste below
- debug:
    msg: '{{ [1, 1, 1, 2, 3, 3, 4, 5, 6, 7, 8, 8, 8, 8, 8, 8] | duplicate }}'

...

MSG:

[1, 3, 8]
```

### CODE REVIEW

File: lib/ansible/plugins/filter/mathstuff.py [hunk 1]
```diff
@@ -45,6 +45,23 @@ def unique(a):
     return c
 
 
+def duplicate(a):
+    if isinstance(a, collections.Hashable):
+        c = set(a)
```
Comment (mkrizek) [inline lib/ansible/plugins/filter/mathstuff.py:91]: But this is `unique` no?

I'd also add tests.

Comment (flatline-studios) [reply lib/ansible/plugins/filter/mathstuff.py:91]: I'm by no means a python expert, but I think line 49 and 50 means, if the argument is a dictionary, then return it without alteration.

As dictionary's aren't hashable, but lists are, and this filter only operates on lists, then this check needs to be made.

As I say I could be wrong on this though.

Ill try and find some time later to knock up a few tests...

**Comment by acozine** [comment]
@flatline-studios you'll need to rebase your branch (instead of merging the most recent `devel` branch into it) before our CI system will kick off a new test run.

**Comment by flatline-studios** [comment]
> @flatline-studios you'll need to rebase your branch (instead of merging the most recent `devel` branch into it) before our CI system will kick off a new test run.

@acozine - I did (at least I thought I did). I followed the instructions at https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html, and pushed the changes back to my repo.

I'm new to git, so I don't quite understand what you mean to be honest.

**Comment by bcoca** [comment]
<strike>isn't this the same as:
``` list1|intersect(list2)|unique ```?</strike> nvmd misread

**Comment by flatline-studios** [comment]
Nope. This is for one list. Not for comparing two. Take a look at the example in the original post.

**Pull request closed without merging**